custom/yeksi_naa_reports/sqldata.py

```diff
@@ -38,6 +39,14 @@ def months(self):
         return [month.date() for month in
                 rrule(freq=MONTHLY, dtstart=self.config['startdate'], until=self.config['enddate'])]
 
+    def date_in_selected_date_range(self, date):
+        return self.months[0] <= date < self.months[-1] + relativedelta(months=1)
+
+    def get_index_of_month_in_selected_data_range(self, date):
+        for index in range(len(self.months)):
+            if date < self.months[index] + relativedelta(months=1):
+                return index
```

Comment:
is it ever possible that this condition would never be hit and this function would return `None`? It seems like a lot of code below would break if that ever happened, so could be good to be explicit (e.g. by throwing an exception outside the bottom of the loop if it reaches there).

Reply:
It will never hit `None` as before each use we check `self.date_in_selected_date_range`, which checks if it would hit `None`

Reply:
got it, thanks